src/progress_bar.rs

```diff
@@ -132,19 +139,35 @@ impl ProgressBar {
     /// When steady ticks are enabled, calling [`ProgressBar::tick()`] on a progress bar does not
     /// have any effect.
     pub fn enable_steady_tick(&self, interval: Duration) {
-        Ticker::spawn(&self.state, interval)
+        if interval.is_zero() {
+            return;
+        }
+
+        self.stop_and_replace_ticker(Some(Ticker::new(interval, &self.state)));
     }
 
     /// Undoes [`ProgressBar::enable_steady_tick()`]
     pub fn disable_steady_tick(&self) {
-        self.state().ticker = None;
+        self.stop_and_replace_ticker(None);
+    }
+
+    fn stop_and_replace_ticker(&self, new_ticker: Option<Ticker>) {
```

Comment:
I think this should take an `interval: Option<Duration>` instead of `Option<Ticker>`, which simplifies the caller. I would also consider defining a `TickerState` which contains the `Mutex<Option<Ticker>>` and see if that cleans things up.

Reply:
Yeah, maybe not, let's leave it for now.